test/models/chan.js

```diff
@@ -27,89 +46,123 @@ describe("Chan", function() {
-				new User({nick: "Max-P", modes: ["o"]}, prefixLookup),
-			]});
-			chan.sortUsers(network);
+			const chan = new Chan();
+			chan.setUser(new User({nick: "JocelynD", modes: ["a", "o"]}, prefixLookup));
+			chan.setUser(new User({nick: "YaManicKill", modes: ["v"]}, prefixLookup));
+			chan.setUser(new User({nick: "astorije", modes: ["h"]}, prefixLookup));
+			chan.setUser(new User({nick: "xPaw", modes: ["q"]}, prefixLookup));
+			chan.setUser(new User({nick: "Max-P", modes: ["o"]}, prefixLookup));
 
 			expect(getUserNames(chan)).to.deep.equal([
 				"xPaw", "JocelynD", "Max-P", "astorije", "YaManicKill",
 			]);
 		});
 
 		it("should sort a mix of users and modes", function() {
-			var chan = new Chan({users: [
-				new User({nick: "JocelynD"}, prefixLookup),
-				new User({nick: "YaManicKill", modes: ["o"]}, prefixLookup),
-				new User({nick: "astorije"}, prefixLookup),
-				new User({nick: "xPaw"}, prefixLookup),
-				new User({nick: "Max-P", modes: ["o"]}, prefixLookup),
-			]});
-			chan.sortUsers(network);
+			const chan = new Chan();
+			chan.setUser(new User({nick: "JocelynD"}, prefixLookup));
+			chan.setUser(new User({nick: "YaManicKill", modes: ["o"]}, prefixLookup));
+			chan.setUser(new User({nick: "astorije"}, prefixLookup));
+			chan.setUser(new User({nick: "xPaw"}, prefixLookup));
+			chan.setUser(new User({nick: "Max-P", modes: ["o"]}, prefixLookup));
 
 			expect(getUserNames(chan)).to.deep.equal(
 				["Max-P", "YaManicKill", "astorije", "JocelynD", "xPaw"]
 			);
 		});
 
 		it("should be case-insensitive", function() {
-			var chan = new Chan({users: ["aB", "Ad", "AA", "ac"].map(makeUser)});
-			chan.sortUsers(network);
+			const chan = new Chan();
+			chan.setUser(new User({nick: "aB"}, prefixLookup));
+			chan.setUser(new User({nick: "Ad"}, prefixLookup));
+			chan.setUser(new User({nick: "AA"}, prefixLookup));
+			chan.setUser(new User({nick: "ac"}, prefixLookup));
 
 			expect(getUserNames(chan)).to.deep.equal(["AA", "aB", "ac", "Ad"]);
 		});
 
 		it("should parse special characters successfully", function() {
-			var chan = new Chan({users: [
-				"[foo", "]foo", "(foo)", "{foo}", "<foo>", "_foo", "@foo", "^foo",
-				"&foo", "!foo", "+foo", "Foo",
-			].map(makeUser)});
-			chan.sortUsers(network);
+			const chan = new Chan();
+			chan.setUser(new User({nick: "[foo"}, prefixLookup));
+			chan.setUser(new User({nick: "]foo"}, prefixLookup));
+			chan.setUser(new User({nick: "(foo)"}, prefixLookup));
+			chan.setUser(new User({nick: "{foo}"}, prefixLookup));
+			chan.setUser(new User({nick: "<foo>"}, prefixLookup));
+			chan.setUser(new User({nick: "_foo"}, prefixLookup));
+			chan.setUser(new User({nick: "@foo"}, prefixLookup));
+			chan.setUser(new User({nick: "^foo"}, prefixLookup));
+			chan.setUser(new User({nick: "&foo"}, prefixLookup));
+			chan.setUser(new User({nick: "!foo"}, prefixLookup));
+			chan.setUser(new User({nick: "+foo"}, prefixLookup));
+			chan.setUser(new User({nick: "Foo"}, prefixLookup));
```

Comment:
For this and all others above, could you keep the condensed version instead?

```js
[
	"[foo", "]foo", "(foo)", "{foo}", "<foo>", "_foo", "@foo", "^foo",
	"&foo", "!foo", "+foo", "Foo",
].forEach((nick) => {
  chan.setUser(new User({nick: nick}, prefixLookup));
});

Reply:
Done.